pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,333 @@
+	// List lists the existing containers that match the ContainerFilter.
+	// The returned list should only include containers previously created
+	// by this ContainerManager.
+	List(filter ContainerFilter) ([]Container, error)
+	// Status returns the status of the container.
+	Status(id RawContainerID) (ContainerStatus, error)
+	// Exec executes a command in the container.
+	Exec(RawContainerID string, cmd []string, streamOpts StreamOptions) error
+}
+
+type ContainerConfig struct {
+	// Name of the container.
+	Name string
+	// Image to use.
+	Image ImageSpec
+	// Command to execute (i.e., entrypoint for docker)
+	Command []string
+	// Args for the Command (i.e., command for docker)
+	Args []string
+	// Current working directory of the command.
+	WorkingDir string
+	// List of environment variable to set in the container
+	Env []KeyValue
+	// Mounts specifies mounts for the container
+	Mounts []Mount
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Privileged runs the container in the privileged mode.
+	Privileged bool
+	// ReadOnlyRootFS sets the root filesystem of the container to be
+	// read-only.
+	ReadOnlyRootFS bool
+	// Path to store the container log on the host (i.e., outside of the
+	// sandbox).
+	LogPath string
+
+	// Variables for interactive containers, these have very specialized
+	// use-cases (e.g. debugging).
+	// TODO: Determine if we need to continue supporting these fields that are
+	// part of Kubernetes's Container Spec.
+	Stdin     bool
+	StdinOnce bool
+	TTY       bool
+
+	// Linux contains configuration specific to Linux containers.
+	Linux *LinuxContainerConfig
+}
+
+// LinuxContainerConfig contains platform-specific configuration for
+// Linux-based containers.
+type LinuxContainerConfig struct {
+	// Resources specification for the container.
+	Resources LinuxContainerResources
+	// AddCapabilities lists capabilities to add.
+	AddCapabilities []string
+	// DropCapabilities lists capabilities to drop.
+	DropCapabilities []string
+	// SELinux is the SELinux context to be applied.
+	SELinux *api.SELinuxOptions
+}
+
+// LinuxContainerResources specifies Linux specific configuration for
+// resources.
+// TODO: Consider using Resources from opencontainers/runtime-spec/specs-go
+// directly.
+type LinuxContainerResources struct {
+	// CPU CFS (Completely Fair Scheduler) period
+	CPUPeriod uint64
+	// CPU CFS (Completely Fair Scheduler) quota
+	CPUQuota uint64
+	// CPU shares (relative weight vs. other containers)
+	CPUShares uint64
+	// Memory limit in bytes
+	MemoryLimitInBytes uint64
+	// OOMScoreAdj adjusts the oom-killer score.
+	OomScoreAdj int
+	// Swappiness specifies how aggressive the kernel will swap memory pages.
+	// Range from 0 to 100.
+	Swappiness uint64
```

Comment:
Removing this field after offline discussion. Since we only want to disable swappiness, it's better not to expose this as part of the config. They'll be included in the documentation and runtime conformance tests